src/components/dashboard/sidebar/OfferSummaryCard.jsx

```diff
@@ -1,38 +1,72 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
+import { Badge } from '@edx/paragon';
 
 import SidebarCard from './SidebarCard';
-import { OFFER_SUMMARY_TITLE, OFFER_SUMMARY_NOTICE } from './data/constants';
-
-const OfferSummaryCard = ({ offersCount, className }) => {
-  const renderCardBody = () => (
-    <>
-      <div className="h1 text-center">
-        {offersCount}
-      </div>
-      <p className="m-0">
-        {OFFER_SUMMARY_NOTICE}
-      </p>
-    </>
+import {
+  OFFER_SUMMARY_TITLE,
+  OFFER_SUMMARY_NOTICE,
+  OFFERS_AVAILABLE_BADGE_VARIANT,
+  OFFERS_AVAILABLE_BADGE_LABEL,
+  COUPON_CODES_REQUESTED_BADGE_VARIANT,
+  COUPON_CODES_REQUESTED_BADGE_LABEL,
+} from './data/constants';
+
+const OfferSummaryCard = ({ offersCount, couponCodeRequestsCount, className }) => {
+  const badgeVariantAndLabel = useMemo(() => {
+    if (offersCount > 0) {
+      return ({
+        variant: OFFERS_AVAILABLE_BADGE_VARIANT,
+        label: OFFERS_AVAILABLE_BADGE_LABEL,
+      });
+    }
+
+    if (couponCodeRequestsCount > 0) {
+      return ({
+        variant: COUPON_CODES_REQUESTED_BADGE_VARIANT,
+        label: COUPON_CODES_REQUESTED_BADGE_LABEL,
+      });
+    }
+
+    return null;
+  }, []);
+
+  const renderCardTitle = () => (
+    <div className="d-flex align-items-end justify-content-between">
```

Comment:
Probably use `align-items-start` here in case the title contents wraps by chance, the badge would stay at the top?